### DISCUSSION TITLE
Can I turn on logging for production?

### DISCUSSION THREAD

**Discussion by changyeamoon** [TanStack/query#4216] [Q&A]
https://tanstack.com/query/v4/docs/guides/migrating-to-react-query-4#logging-in-production

I see v4 no longer logs in production, but I need it because my development environment has to be in production mode (salesforce use case)

In the same vein, is there a way to show the QueryDevTool in production too?

**Comment by TkDodo** [comment]
devtools in production: https://tanstack.com/query/v4/docs/devtools#devtools-in-production

logging in production was removed in v4.

**Reply by changyeamoon** [reply]
thanks for the devtool link 👍🏼 

and Yes I know, which is why im asking if theres a way to optionally turn it on

**Reply by TkDodo** [reply]
no, there is not. the logs are not really helpful in production - we are logging hints towards potential bugs that should be addressed during development - like queryKeys not being an array.

it's similar to the logs that react gives you as well during development - like the warning you get when you have a duplicate key on a list. They also don't exist during production.
